controller.h

```diff
@@ -0,0 +1,494 @@
+#pragma once
+using namespace std;
+
+// Controller
+//
+// Represents the entire system, contains the canvases which contain the features which contain the effects.
+
+#include "json.hpp"
+#include "interfaces.h"
+#include "basegraphics.h"
+#include "effectsmanager.h"
+#include <vector>
+#include "global.h"
+#include "canvas.h"
+#include "ledfeature.h"
+#include "serialization.h"
+#include "effects/colorwaveeffect.h"
+#include "effects/starfield.h"
+#include "effects/videoeffect.h"
+#include "effects/misceffects.h"
+#include "palette.h"
+#include "effects/paletteeffect.h"
+#include "effects/fireworkseffect.h"
+
+class Controller;
+inline void from_json(const nlohmann::json &j, unique_ptr<Controller> & ptrController);
+
+class Controller : public IController
+{
+  private:
+
+    vector<unique_ptr<ICanvas>> _canvases;
+    uint16_t                    _port;
+
+  public:
+
+    Controller(uint16_t port = 7777) : _port(port)
+    {
+    }
+
+    Controller() : _port(7777) 
+    {
+    }
+
+    vector<reference_wrapper<ICanvas>> Canvases() const override
+    {
+        vector<std::reference_wrapper<ICanvas>> canvases;
+        canvases.reserve(_canvases.size());
+        for (const auto& canvas : _canvases)
+            canvases.push_back(*canvas);
+        return canvases;
+    }
+
+    static std::unique_ptr<Controller> CreateFromFile(const std::string& filePath) 
```

Comment:
...and that effectively kills the idea I had to easily override the standard set of canvases to match my one-Mesmerizer setup. I'll go back to the drawing board on that one. :)